gh_test.go

```diff
@@ -57,24 +58,24 @@ func TestRESTClient(t *testing.T) {
 	os.Setenv("GH_CONFIG_DIR", tempDir)
 	os.Setenv("GH_TOKEN", "GH_TOKEN")
 
-	http := httpmock.NewRegistry(t)
-	http.Register(
-		httpmock.REST("GET", "some/test/path"),
-		httpmock.StatusStringResponse(200, `{"message": "success"}`),
-	)
+	gock.New("https://api.github.com").
```

Comment:
Instead of `gock.New()`, could we use a gock initialization method that doesn't intercept the default `http.DefaultTransport`, but makes a new transport that we can pass via options to `RESTClient()`? That way it's more clear what's being initialized, there is no reliance to global state, and we don't have to do `t.Cleanup(gock.Off())`

Reply:
I thought we could get around this but [this check](https://github.com/h2non/gock/blob/48c3ba5c871a49c9576f33d5c7d88eff9404d757/transport.go#L48) functionality deep in the `gock` implementation makes it infeasible.

Reply:
Ah I see; it looks like the library is designed in a way that it always wants to be used with global intercept. 👍